mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala

```diff
@@ -220,20 +240,12 @@ private[spark] object BLAS extends Serializable {
       case sx: SparseVector =>
         f2jBLAS.dscal(sx.values.length, a, sx.values, 1)
```

Comment:
I think you can apply the optimization here too; just depends on the length of sx.values

Reply:
done.